lib/prop/leaky_bucket_strategy.rb

```
@@ -5,40 +5,73 @@
 module Prop
   class LeakyBucketStrategy
     class << self
+      def _throttle_leaky_bucket(handle, key, cache_key, options)
+        (over_limit, bucket) = options.key?(:decrement) ?
```

Should we raise an error here in case increment is less than 1?

---

added some error handling in the validate method